connection.go

```diff
@@ -122,83 +118,77 @@ func (c *Conn) BeginTx(ctx context.Context, opts driver.TxOptions) (driver.Tx, e
 		return nil, errors.Join(errBeginTx, errIsolationLevelNotSupported)
 	}
 
-	if _, err := c.ExecContext(ctx, `BEGIN TRANSACTION`, nil); err != nil {
+	if _, err := conn.ExecContext(ctx, `BEGIN TRANSACTION`, nil); err != nil {
 		return nil, err
 	}
 
-	c.tx = true
-	return &tx{c}, nil
+	conn.tx = true
+	return &tx{conn}, nil
```

Comment:
nit: Newline before return (there are multiple occurrences).

```suggestion
	conn.tx = true

	return &tx{conn}, nil
```